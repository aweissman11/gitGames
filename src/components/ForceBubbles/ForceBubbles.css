.all-bubbles {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.force-container {
  position: static;
  width: 100vw;
  height: 100vh;
  /* margin: auto; */
  position: unset;
}

.force-box {
  position: relative;
  top: 0;
  width: 100vw;
  float: left;
  height: 100%;
  overflow: unset;
}

.force-bubbles {
  z-index: 999;
}


