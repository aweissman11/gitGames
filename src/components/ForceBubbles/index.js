import React, {
  Component
} from 'react';
import * as d3 from 'd3';

import './ForceBubbles.css';

class ForceBubbles extends Component {
  componentDidMount() {
		const width = 960
    const height = this.props.customHeight
    const nodes = d3.range(400).map(function() { return {r: Math.random() * 12 + 4, s:"hello"}; }),
        root = nodes[0];
    
    const sequentialScale = d3.scaleSequential()
      .domain([0, 100])
      .interpolator(d3.interpolatePlasma);

    root.radius = 0;
    root.fixed = true;

    const forceX = d3.forceX(width / 2).strength(0.015)
    const forceY = d3.forceY(height / 2).strength(0.015)

    const force = d3.forceSimulation()
    .velocityDecay(0.2)
    .force("x", forceX)
    .force("y", forceY)
    .force("collide", d3.forceCollide().radius(function(d){
      if(d === root){
        return Math.random() * 50 +100;
      }
      return d.r + 0.5;
    }).iterations(5)) 
    .nodes(nodes).on("tick", ticked);

    const svg = d3.select(".force-bubbles")
    .attr("width", width)
    .attr("height", height);

    svg.selectAll("circle")
        .data(nodes.slice(1))
        .enter().append("circle")
        .attr("r", function(d) { return d.r; }) 
        .style("fill", function(d, i) { return sequentialScale(i % 100); }); 

    function ticked(e) {
        svg.selectAll("circle")
            .attr("cx", function(d) { return d.x; })
            .attr("cy", function(d) { return d.y; });
    };

    svg.on("mousemove", function() {
        var p1 = d3.mouse(this);
        root.fx = p1[0];
        root.fy = p1[1];
        force.alphaTarget(0.5).restart();
    });
}

  render() {
    return (
      <div className="force-container">
        <div className="force-box">
          <svg className="force-bubbles" />
        </div>
      </div>
    );
  }
}

export default ForceBubbles;