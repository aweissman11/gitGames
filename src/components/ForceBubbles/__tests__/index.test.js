import React from 'react';
import { shallow } from 'enzyme';

import ForceBubbles from '../index';

describe('ForceBubbles', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ForceBubbles />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
