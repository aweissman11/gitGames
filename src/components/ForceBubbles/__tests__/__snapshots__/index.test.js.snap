// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForceBubbles should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForceBubbles />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="force-box"
      >
        <svg
          className="force-bubbles"
        />
      </div>,
      "className": "force-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          className="force-bubbles"
        />,
        "className": "force-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "force-bubbles",
        },
        "ref": null,
        "rendered": null,
        "type": "svg",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="force-box"
        >
          <svg
            className="force-bubbles"
          />
        </div>,
        "className": "force-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="force-bubbles"
          />,
          "className": "force-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "force-bubbles",
          },
          "ref": null,
          "rendered": null,
          "type": "svg",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
